### PR TITLE
Implemented getpeername and getsockname calls.

### PR DESCRIPTION
**Pull request by kapkic**

### CODE REVIEW

File: src/trusted/service_runtime/nacl_syscall_common.c [hunk 1]
```diff
@@ -4897,3 +4897,69 @@ int32_t NaClSysStatfs(struct NaClAppThread *natp,
...
+
+  struct sockaddr * sysaddr = (struct sockaddr*) NaClUserToSysAddrRange(nap, (uintptr_t) addr, sizeof(struct sockaddr));
+  socklen_t * sysaddrlen = (socklen_t*) NaClUserToSysAddrRange(nap, (uintptr_t) addrlen, sizeof(socklen_t));
+
+  if ((void*) kNaClBadAddress == sysaddr) {
+    NaClLog(2, "NaClSysGetpeername could not translate socket address, returning %d\n", -NACL_ABI_EFAULT);
+    return -NACL_ABI_EFAULT;
+  }
+
+    if ((void*) kNaClBadAddress == sysaddrlen) {
```
Comment (jesings) [inline src/trusted/service_runtime/nacl_syscall_common.c]: funky indentation

File: src/trusted/service_runtime/nacl_syscall_common.c [hunk 2]
```diff
@@ -4897,3 +4897,69 @@ int32_t NaClSysStatfs(struct NaClAppThread *natp,
...
```
Comment (jesings) [inline src/trusted/service_runtime/nacl_syscall_common.c]: Also funky indentation

**Review by jesings — APPROVED** [review]
Looks pretty much good, needs the merge and a couple of tiny fixes

File: src/shared/platform/lind_platform.c [hunk 3]
```diff
@@ -790,3 +780,25 @@ int lind_socket (int domain, int type, int protocol, int cageid)
...
     LIND_API_PART3;
 }
+
+int lind_getsockname (int sockfd, struct sockaddr * addr, socklen_t* addrlen, int cageid)
+{
+    LIND_API_PART1;
+    callArgs = Py_BuildValue("(i[iii])", LIND_safe_net_getsockname, sockfd, *addrlen, cageid);
+    LIND_API_PART2;
+    if (*addrlen > _len) {COPY_DATA(addr, _len);}
+    else {COPY_DATA(addr, _len);}
```
Comment (jesings) [inline src/shared/platform/lind_platform.c]: Also the else case should be *addrlen, not _len, same for getpeername

Comment (kapkic) [reply src/shared/platform/lind_platform.c]: Fixed.

File: src/trusted/service_runtime/nacl_syscall_common.c (hunk 1 above)
Comment (kapkic) [reply src/trusted/service_runtime/nacl_syscall_common.c]: Fixed.

File: src/trusted/service_runtime/nacl_syscall_common.c (hunk 2 above)
Comment (kapkic) [reply src/trusted/service_runtime/nacl_syscall_common.c]: Fixed.

**Comment by rennergade** [comment]
Looks good!

**Pull request merged**